src/Package/Impl/DataInspect/DataImport/ImportDataWindow.xaml.cs

```diff
@@ -102,89 +105,63 @@ public ImportDataWindow(string filePath, string name)
-                }
-                // TODO: map names that R uses to actual encoding names
-                try {
-                    return Encoding.GetEncoding(encoding);
-                } catch (ArgumentException) { }
-            }
-            return Encoding.Default;
-        }
-
         private void FileOpenButton_Click(object sender, RoutedEventArgs e) {
-            string filePath = VsAppShell.Current.BrowseForFileOpen(IntPtr.Zero, "CSV file|*.csv");
+            string filePath = VsAppShell.Current.BrowseForFileOpen(IntPtr.Zero, Package.Resources.CsvFileFilter);
             if (!string.IsNullOrEmpty(filePath)) {
-                SetFilePathAsync(filePath).DoNotWait();
+                SetFilePath(filePath);
             }
         }
 
-        private async Task SetFilePathAsync(string filePath, string name = null) {
+        private void SetFilePath(string filePath, string name = null) {
             FilePathBox.Text = filePath;
             FilePathBox.CaretIndex = FilePathBox.Text.Length;
             FilePathBox.ScrollToEnd();
 
             VariableNameBox.Text = name ?? Path.GetFileNameWithoutExtension(filePath);
-            await PreviewAsync();
+            PreviewContent();
         }
 
-        private async void RunButton_Click(object sender, RoutedEventArgs e) {
-            var expression = BuildCommandLine(false);
-            if (expression != null) {
-                RunButton.IsEnabled = CancelButton.IsEnabled = false;
-                // TODO: this may take a while and must be cancellable
-                await RunAsync(expression);
-                RunButton.IsEnabled = CancelButton.IsEnabled = true;
-            }
-        }
-
-        private void CancelButton_Click(object sender, RoutedEventArgs e) {
-            Close();
-        }
-
-        private async Task PreviewAsync() {
-            if(_encodingsInitTask == null) {
+        private async void PreviewContent() {
+            if (string.IsNullOrEmpty(FilePathBox.Text)) {
                 return;
             }
 
-            await _encodingsInitTask;
-
-            if (!string.IsNullOrEmpty(FilePathBox.Text)) {
-                string text = ReadFile(FilePathBox.Text, ToEncoding(GetSelectedString(EncodingComboBox)));
-                InputFilePreview.Text = text;
-            }
-
-            var expression = BuildCommandLine(true);
-            if (expression != null) {
-                try {
-                    var grid = await GridDataSource.GetGridDataAsync(expression, null);
-
-                    PopulateDataFramePreview(grid);
-                    ErrorBlock.Visibility = Visibility.Collapsed;
-                    DataFramePreview.Visibility = Visibility.Visible;
-                } catch (Exception ex) {
-                    OnError(ex.Message);
+            int cp = GetSelectedValueAsInt(EncodingComboBox);
+            PreviewFileContent(FilePathBox.Text, cp);
+            await ConvertToUtf8(FilePathBox.Text, cp, MaxPreviewLines);
```

Comment:
Bad idea awaiting inside void method. If you want to insure that async method is executed in UI thread, use `ICoreShell.DispatchOnMainThreadAsync`